deno/payloads/v8/channel.ts

```diff
@@ -32,35 +32,82 @@ export interface APIPartialChannel {
 }
 
 /**
- * https://discord.com/developers/docs/resources/channel#channel-object-channel-structure
+ * This interface is used to allow easy extension for other channel types. While
+ * also allowing `APIPartialChannel` to be used without breaking.
  */
-export interface APIChannel extends APIPartialChannel {
+export interface APIChannelBase<T extends ChannelType> extends APIPartialChannel {
+	type: T;
+}
+
+export type TextChannelTypes = ChannelType.DM | ChannelType.GroupDM | ChannelType.GuildNews | ChannelType.GuildText;
+
+// TODO: remove when text in voice is released
```

Comment:
I'm dumb as heck, and @SpaceEEC enlightened me! The GuildChannelTypes shouldn't be removed haha. The TODO should be added to TextChannelTypes that we can soon add voice channels to it